crates/matrix-sdk-crypto/src/gossiping/machine.rs

```diff
@@ -539,13 +539,15 @@ impl GossipMachine {
         device: &Device,
         content: SecretSendContent,
     ) -> OlmResult<Session> {
-        let event_type = content.event_type();
-        let (used_session, content) = device.encrypt(event_type, content).await?;
+        let event_type = content.event_type().to_owned();
+        let (used_session, content) = device.encrypt(&event_type, content).await?;
+
+        let event_type = content.event_type().to_owned();
```

Comment:
this line looks redundant?

Reply:
At first glance one would think so. But the first `event_type` is for `content` while the second `event_type` is for `content` after the `device.encrypt()` step.

Perhaps renaming the second `content` to `encrypted_content` might make this clearer.

Reply:
oh right, yes. Some different names might help.